### PR TITLE
*: migrate dependency management to Go1.11 module

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Fix https://github.com/pingcap/tidb/issues/7922

### What is changed and how it works?

* `go mod init`
* remove `vendor` directory
* rename import path "github.com/pkg/errors" to "github.com/pingcap/errors"
* tiny update `util/logutil` to fix CI

### Check List 

Tests 

 - No code

Side effects

- Don't forget to update other repos

@disksing @lysu

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,15 +54,15 @@ all: dev server benchkv
...
+dev: checklist check test
 
 build:
 	$(GOBUILD)
 
 # The retool tools.json is setup from hack/retool-install.sh
 check-setup:
 	@which retool >/dev/null 2>&1 || go get github.com/twitchtv/retool
-	@retool sync
+	@GO111MODULES=off retool sync
```
Comment (disksing) [inline Makefile]: GO111MODULE

**Comment by tiancaiamao** [comment]
I nearly make all test pass, but after merge master and resolve conflicts, I find tidb-tools have introduce module and [it use a `pd` version conflict with here](https://github.com/pingcap/tidb-tools/pull/96).

**Comment by gregwebs** [comment]
One thing that can help avoid disrupting workflows is to do the switch but to use go mod's vendoring support (we did this in pd).
Even if we don't want to commit that, it is very helpful for code review.

**Comment by tiancaiamao** [comment]
Good suggestion, we can adopt it in tidb-tools. @gregwebs 
vendor is not included here because:
1. tidb update parser frequently, and that will make this repo expansion quickly
2. vendor is not recommend by the Go team

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	golang.org/x/net v0.0.0-20181029044818-c44066c5c816
+	golang.org/x/oauth2 v0.0.0-20181031022657-8527f56f7107 // indirect
+	golang.org/x/sys v0.0.0-20181030150119-7e31e0c00fa0 // indirect
+	golang.org/x/text v0.3.0
+	golang.org/x/time v0.0.0-20180412165947-fbb02b2291d2 // indirect
+	golang.org/x/tools v0.0.0-20181030221726-6c7e314b6563 // indirect
+	google.golang.org/appengine v1.2.0 // indirect
+	google.golang.org/genproto v0.0.0-20181029155118-b69ba1387ce2 // indirect
+	google.golang.org/grpc v1.16.0
+	gopkg.in/airbrake/gobrake.v2 v2.0.9 // indirect
```
Comment (gregwebs) [inline go.mod]: What is this for?

**Comment by gregwebs** [comment]
I think go.mod needs a more thorough review still. We may have a repeat of the same issue we had in PD of pulling in some odd dependencies https://github.com/pingcap/pd/pull/1269#pullrequestreview-163861009
Talk with @disksing

**Comment by tiancaiamao** [comment]
Yeah, the hardest step is resolving the dependency and make CI work.
https://github.com/pingcap/tidb/issues/7922

Now it's time to polish it, feel free to send some PRs. @gregwebs

**Comment by gregwebs** [comment]
I won't be sending PRs any time soon. You should be able to ask @disksing what he did.

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	github.com/Shopify/sarama v1.19.0 // indirect
+	github.com/Shopify/toxiproxy v2.1.3+incompatible // indirect
+	github.com/apache/thrift v0.0.0-20161221203622-b2a4d4ae21c7 // indirect
+	github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973 // indirect
+	github.com/blacktear23/go-proxyprotocol v0.0.0-20171102103907-62e368e1c470
+	github.com/chzyer/readline v0.0.0-20171208011716-f6d7a1f6fbf3 // indirect
+	github.com/cockroachdb/cmux v0.0.0-20170110192607-30d10be49292 // indirect
+	github.com/codahale/hdrhistogram v0.0.0-20161010025455-3a0bb77429bd // indirect
+	github.com/coreos/bbolt v1.3.1-coreos.6 // indirect
+	github.com/coreos/etcd v3.3.10+incompatible
```
Comment (disksing) [inline go.mod]: Why there is an `incompatible`? Does it mean that we have dependencies that import different versions of etcd?

**Comment by disksing** [comment]
@gregwebs In fact, in the case where the vendor is deleted, it is difficult to know the difference between the dependencies used by go.mod and the previous vendor dependencies.

**Pull request merged**